### PR TITLE
OTP all three pages 

### PR DESCRIPTION
**Pull request by dbayly-freshworks**
Completed all three pages for the otp flow. 

Screen Shot 2021-05-13 at 4 02 10 PM
Screen Shot 2021-05-13 at 3 43 37 PM
Screen Shot 2021-05-13 at 3 43 18 PM

### CODE REVIEW

File: client/src/pages/public/Otp.js [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+    },
+  };
+});
+
+export default (props) => {
+  const [pageState, setPageState] = useState('default');
+  const classes = useStyles();
+  // Get the query string from the url
+  // Remove the questionmark from the search
+  const query = qs.parse(props.location.search.slice(1), '&', '=');
```
Comment (Zenen-freshworks) [inline client/src/pages/public/Otp.js:52]: Is importing location as a prop necessary here? I've used window.location in the past.

Comment (dbayly-freshworks) [reply client/src/pages/public/Otp.js:52]: I played around with both. `window.location` gave me the full url, where props.location gave me a more detailed object with just the `search` as one of the options.  

I felt this was cleaner.

Comment (Zenen-freshworks) [reply client/src/pages/public/Otp.js:52]: Cool, I'm on board with clean code!

**Review by Zenen-freshworks — APPROVED** [review]

File: client/src/pages/public/Otp.js (hunk 1 above)
Comment (arranfw) [reply client/src/pages/public/Otp.js:52]: Just to add to the conversation, I'd vouch for react router + native classes like in this example, not necessary to change anything just tossing in my 2c :p

https://reactrouter.com/web/example/query-parameters

File: client/src/routes/index.js [hunk 2]
```diff
@@ -104,6 +105,7 @@ export default () => {
               component={ParticipantConfirmation}
             />
             <Route exact path={Routes.Base} component={ParticipantForm} />
+            <Route exact path={Routes.OtP} component={OTP} />
```
Comment (arranfw) [inline client/src/routes/index.js:108]: Can we name this something along the lines of `confirm-interest`? I don't feel OTP is descriptive enough

**Review by arranfw — COMMENTED** [review]
General comment about making the code more succinct, I would make an object with the different strings/icons/links as values and the query param value as keys and then just create one component which takes in that data as props. This would help to reduce the repetition a bit.

File: client/src/routes/index.js (hunk 2 above)
Comment (dbayly-freshworks) [reply client/src/routes/index.js:108]: Sounds good. 👍

File: client/src/pages/public/Otp.js [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+        )}
+        {pageState === 'success' && (
+          <Box display='flex' flexDirection='column'>
+            <CheckBoxIcon className={classes.check} fontSize={'large'} />
+            <Typography className={classes.title} variant={'h3'} align={'center'}>
+              Thanks For Letting Us Know
+            </Typography>
+            <Typography variant={'body1'} align={'center'}>
+              You have succesfully confirmed your interest. To view your current status, you can log
+              into the HCAP application
```
Comment (arranfw) [inline client/src/pages/public/Otp.js:87]: Realizing this copy isn't exactly accurate, we don't have a login system

Comment (dbayly-freshworks) [reply client/src/pages/public/Otp.js:87]: Thats a good point.
I'll put in placeholder text for now and bring this up in standup.

File: server/migrations/1613607352100_withdraw-duplicate-participants.js [hunk 4]
```diff
@@ -3,11 +3,56 @@ const { dbClient, collections } = require('../db');
 
 exports.up = async () => {
   const duplicates = [
-    300616, 300598, 300105, 300219, 300141, 300153, 300217, 300195, 300180, 300173, 300199, 300246,
```
Comment (dbayly-freshworks) [inline server/migrations/1613607352100_withdraw-duplicate-participants.js:6]: I ran the formatter and this got caught in the crossfire

**Review by arranfw — APPROVED** [review]

File: client/src/constants/routes.js [hunk 5]
```diff
@@ -10,6 +10,7 @@ export default Object.freeze({
   ParticipantForm: '/participant-form',
   ParticipantConfirmation: '/participant-confirmation',
   EmployerConfirmation: '/employer-confirmation',
+  ConfirmInterest: '/participant-otp',
```
Comment (arranfw) [inline client/src/constants/routes.js:13]:
```suggestion
  ConfirmInterest: '/confirm-interest',
```

File: client/src/constants/confirmInterestConstants.js [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+  hasButton: false,
+  buttonText: '',
+  iconClass: 'check',
+};
+export const confirmInterestError = {
+  title: 'Expired Link',
+  icon: InfoIcon,
+  description: `This link is invalid or your interest has already been confirmed.`,
+  hasButton: false,
+  buttonText: '',
```
Comment (arranfw) [inline client/src/constants/confirmInterestConstants.js:26]: Why don't we set buttonText to null when we don't want to display it? The extra attribute seems unnecessary

File: client/src/pages/public/ConfirmInterest.js [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+        <Box display='flex' flexDirection='column'>
+          <Icon component={state.icon} className={classes[state.iconClass]} fontSize={'large'} />
+          <Typography className={classes.title} variant={'h3'} align={'center'}>
+            {state.title}
+          </Typography>
+          <Typography variant={'body1'} align={'center'}>
+            {state.description}
+          </Typography>
+          <Box className={classes.button}>
+            {state.hasButton ? <Button text={state.buttonText} onClick={handleCheckToken} /> : null}
```
Comment (arranfw) [inline client/src/pages/public/ConfirmInterest.js:74]: Like above, let's just check if buttonText exists

File: client/src/constants/confirmInterestConstants.js [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+export const confirmInterestSuccess = {
+  title: 'Thanks For Letting Us Know',
+  icon: CheckBoxIcon,
+  description: `You have successfully confirmed your continued interest in participating in the Health Career Access Program (HCAP). Your information has been updated accordingly.`,
+  hasButton: false,
+  buttonText: '',
+  iconClass: 'check',
+};
+export const confirmInterestError = {
+  title: 'Expired Link',
```
Comment (arranfw) [inline client/src/constants/confirmInterestConstants.js:22]: Just confirming this title was finalized? It doesn't seem to account for invalid links like the description does

Comment (dbayly-freshworks) [reply client/src/constants/confirmInterestConstants.js:22]: Looks like I missed this one, correct wording is `Invalid Link`

**Review by arranfw — APPROVED** [review]

**Review by ashtonmeuser — APPROVED** [review]

**Pull request merged**